core/src/main/java/fj/function/Booleans.java

```diff
@@ -92,6 +84,56 @@ public static boolean and(final List<Boolean> l) {
     return Monoid.conjunctionMonoid.sumLeft(l);
   }
 
+    /**
+     * maps given function to the predicate function
+     * @param p predicate to be mapped over
+     * @param f function
+     * @return predicate function
+     */
+    public static <A, B>  F<B, Boolean> contramap(F<B, A> f, F<A, Boolean> p){
+        return compose(p, f);
+    }
+
```

Comment:
technically it's a `contramap`, because it's applying function to input of `F&lt;A, Bool&gt;` (negative position)